src/nvim/os/env.c

```diff
@@ -427,6 +428,37 @@ char *vim_getenv(const char *name)
     return xstrdup(kos_env_path);
   }
 
+  // XDG Base Directory expansion.
+  if (strncmp(name, "XDG_", 4) == 0) {
+    const char *home = vim_getenv("HOME");
+    if (!home) {
+      return NULL;
+    }
+
+    const char *home_suffix = "";
+    if (strcmp(name, "XDG_DATA_HOME") == 0) {
+      home_suffix = "/.local/share/";
+    } else if (strcmp(name, "XDG_CONFIG_HOME") == 0) {
+      home_suffix = "/.config/";
+    } else if (strcmp(name, "XDG_CACHE_HOME") == 0) {
+      home_suffix = "/.cache/";
+    } else if (strcmp(name, "XDG_RUNTIME_DIR") == 0) {
+      return xstrdup("/tmp/");  // Arbitrary value, any idea?
+    } else if (strcmp(name, "XDG_CONFIG_DIRS") == 0) {
+      return xstrdup("/etc/xdg");
+    } else if (strcmp(name, "XDG_DATA_DIRS") == 0) {
+      return xstrdup("/usr/local/share/:/usr/share/");
```

Comment:
Anybody knows why here are trailing slashes and above are not? (I have copied defaults from the specification, so don’t say “because you wrote that”.)